src/mtcp/restore_libc.c

```diff
@@ -477,12 +490,22 @@ TLSInfo_SetThreadSysinfo(void *sysinfo)
 void
 TLSInfo_VerifyPidTid(pid_t pid, pid_t tid)
 {
+  char *ptr;
+  long major = strtol(gnu_get_libc_version(), &ptr, 10);
+  long minor = strtol(ptr+1, NULL, 10);
+  ASSERT (major == 2);
+
   pid_t tls_pid, tls_tid;
   char *addr = (char *)get_tls_base_addr();
 
   tls_pid = *(pid_t *)(addr + TLSInfo_GetPidOffset());
   tls_tid = *(pid_t *)(addr + TLSInfo_GetTidOffset());
 
+  // For glibc > 2.24, pid field is unused.
+  if (tls_pid == 0 && minor >= 24) {
```

Comment:
Don't you want the full condition `tls_pid == NULL && major == 2 && minor >= 24`, similarly to what you test for at line 283 above?  If we rely on the earlier `ASSERT (major == 2)`, then the condition to test here requires reading some code from elsewhere.